Sources/Events/EventsManager.swift

```diff
@@ -114,6 +115,39 @@ actor EventsManager: EventsManagerType {
     }
     #endif
 
+    func flushEventsWithBackgroundTask() async {
+        #if os(iOS) || os(tvOS) || VISION_OS
+        let endBackgroundTask = await self.beginBackgroundTask(named: "com.revenuecat.flushEvents")
+        defer {
+            endBackgroundTask?()
+        }
+        #endif
+
+        do {
+            _ = try await self.flushEvents(batchSize: Self.defaultEventBatchSize)
+        } catch {
+            Logger.error(Strings.paywalls.event_flush_failed(error))
+        }
+    }
+
+    func flushFeatureEventsWithBackgroundTask() async {
```

Comment:
Hmm so I understand we removed the `flushAdEvents` since it was only used internally, but I think the current naming is very confusing, as in, `flushEventsWithbackgroundTask` is for ads + feature events and `flushFeatureEventsWithBackgroundTask` is for only feature events... I'm inclining towards separating both flushes, and just call both on the call sites, if we need to split them up. Alternatively, we could rename 
`flushEventsWithbackgroundTask` to `flushAllEventsWithbackgroundTask` so at least it's clearer there are other event types being flushed aside from feature events? cc @polmiro since you worked on this part recently.

Also, some docs on the difference between both could help.